File: solution-java/014_longest_common_prefix.java
/*
	014. Longest Common Prefix
	@name: li jin
	@date: Feb 23, 2016
	@link: https://leetcode.com/problems/longest-common-prefix/
	@time: 1 ms
*/

public class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 0) return "";
        String ans = strs[0];
        int n = ans.length();
        for (String s : strs) {
            while (n > 0 && !s.startsWith(ans)) {
                n--;
                ans = ans.substring(0, n);
            }
            if (n == 0) return "";
        }
        return ans;
    }
}

File: solution-python/014_longest_common_prefix.py
"""
	014. Longest Common Prefix
	@name: li jin
	@date: Feb 23, 2016
	@link: https://leetcode.com/problems/longest-common-prefix/
	@time: 44 ms
"""

class Solution:
    # @return a string
    def longestCommonPrefix(self, strs):
        if len(strs) == 0:
            return ""
        res = strs[0]
        for s in strs:
            while not s.startswith(res):
                if res:
                    res = res[:-1]
                else:
                    return ""
        return res
